homeassistant/components/sensor/mqtt.py

```diff
@@ -104,6 +111,20 @@ def message_received(topic, payload, qos):
                 self._expiration_trigger = async_track_point_in_utc_time(
                     self.hass, self.value_is_expired, expiration_at)
 
+            if self._json_attributes:
+                self._attributes = {}
+                try:
+                    json_dict = json.loads(payload)
+                    if isinstance(json_dict, dict):
+                        attrs = {k: json_dict[k] for k in self._json_attributes
```

Comment:
And while we're at it, another performance nitpick: make `self._json_attributes` a set and then use:
`attrs = {k : json_dict[k] for k in self._json_attributes & json_dict.keys()}`

Reply:
Done, will upload in next commit